File: www/gauge-dual/demo-grid-light.m

//
//  ViewController.m
//  Highcharts
//
//  License: www.highcharts.com/license
//  Copyright © 2016 Highsoft AS. All rights reserved.
//

#import <UIKit/UIKit.h>
#import <Highcharts/Highcharts.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    HIGChartView *chartView = [[HIGChartView alloc] initWithFrame:self.view.bounds];

    chartView.theme = @"grid-light";

    chartView.options = @{
                            @"chart": @{
                                @"type": @"gauge",
                                @"alignTicks": @false,
                                @"plotBackgroundColor": [NSNull null],
                                @"plotBackgroundImage": [NSNull null],
                                @"plotBorderWidth": @0,
                                @"plotShadow": @false
                            },
                            @"title": @{
                                @"text": @"Speedometer with dual axes"
                            },
                            @"pane": @{
                                @"startAngle": @-150,
                                @"endAngle": @150
                            },
                            @"yAxis": @[
                                @{
                                    @"min": @0,
                                    @"max": @200,
                                    @"lineColor": @"#339",
                                    @"tickColor": @"#339",
                                    @"minorTickColor": @"#339",
                                    @"offset": @-25,
                                    @"lineWidth": @2,
                                    @"labels": @{
                                        @"distance": @-20,
                                        @"rotation": @"auto"
                                    },
                                    @"tickLength": @5,
                                    @"minorTickLength": @5,
                                    @"endOnTick": @false
                                },
                                @{
                                    @"min": @0,
                                    @"max": @124,
                                    @"tickPosition": @"outside",
                                    @"lineColor": @"#933",
                                    @"lineWidth": @2,
                                    @"minorTickPosition": @"outside",
                                    @"tickColor": @"#933",
                                    @"minorTickColor": @"#933",
                                    @"tickLength": @5,
                                    @"minorTickLength": @5,
                                    @"labels": @{
                                        @"distance": @12,
                                        @"rotation": @"auto"
                                    },
                                    @"offset": @-20,
                                    @"endOnTick": @false
                                }
                            ],
                            @"series": @[
                                @{
                                    @"name": @"Speed",
                                    @"data": @[
                                        @80
                                    ],
                                    @"tooltip": @{
                                        @"valueSuffix": @" km/h"
                                    }
                                }
                            ]
                        };
    
    [self.view addSubview:chartView];
}

@end

File: www-swift/gauge-dual/demo-grid-light.swift
import Highcharts
import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()

    let chartView = HIChartView(frame: view.bounds)
    chartView.theme = "grid-light"

    let options = HIOptions()

    let chart = HIChart()
    chart.type = "gauge"
    chart.alignTicks = false
    chart.plotBorderWidth = 0
    chart.plotShadow = HICSSObject()
    chart.plotShadow.width = 0
    options.chart = chart

    let title = HITitle()
    title.text = "Speedometer with dual axes"
    options.title = title

    let pane = HIPane()
    pane.startAngle = -150
    pane.endAngle = 150
    options.pane = pane

    let yAxis1 = HIYAxis()
    yAxis1.min = 0
    yAxis1.max = 200
    yAxis1.lineColor = HIColor(hexValue: "339")
    yAxis1.tickColor = HIColor(hexValue: "339")
    yAxis1.minorTickColor = HIColor(hexValue: "339")
    yAxis1.offset = -25
    yAxis1.lineWidth = 2
    yAxis1.labels = HILabels()
    yAxis1.labels.distance = -20
    // yAxis1.labels.rotation = "auto"
    yAxis1.tickLength = 5
    yAxis1.minorTickLength = 5
    yAxis1.endOnTick = false

    let yAxis2 = HIYAxis()
    yAxis2.min = 0
    yAxis2.max = 124
    yAxis2.tickPosition = "outside"
    yAxis2.lineColor = HIColor(hexValue: "933")
    yAxis2.lineWidth = 2
    yAxis2.minorTickPosition = "outside"
    yAxis2.tickColor = HIColor(hexValue: "933")
    yAxis2.minorTickColor = HIColor(hexValue: "933")
    yAxis2.tickLength = 5
    yAxis2.minorTickLength = 5
    yAxis2.labels = HILabels()
    yAxis2.labels.distance = 12
    // yAxis2.labels.rotation = "auto"
    yAxis2.offset = -20
    yAxis2.endOnTick = false

    options.yAxis = [yAxis1, yAxis2]

    let speed = HIGauge()
    speed.name = "Speed"
    speed.data = [80]

    let dataLabels = HIDataLabels()
    dataLabels.formatter = HIFunction(jsFunction: "function () { var kmh = this.y, mph = Math.round(kmh * 0.621); return '<span style=\"color:#339\">' + kmh + ' km/h</span><br/>' + '<span style=\"color:#933\">' + mph + ' mph</span>'; }")
    speed.dataLabels = [dataLabels]

    speed.tooltip = HITooltip()
    speed.tooltip.valueSuffix = " km/h"

    options.series = [speed]

    chartView.options = options

    self.view.addSubview(chartView)
  }

}
